docs/architecture-overview.md

```diff
@@ -0,0 +1,124 @@
+
+## Context
+
+- prev arch
```

Comment:
I'd recommend skipping this part. By the end of this sprint (or the next), we won't have the previous architecture anymore. Describing it here will introduce the reader to two different architectures and significant complexity. This will make it much harder to understand and we want this to be easy for new contributors. I'd focus on the current architecture only / how it works now, and future work.

Reply:
Agreed, I'll pull that out.  Thanks for the suggestion!